### ISSUE TITLE
Allways active?

### ISSUE THREAD

**Issue by tetoNidan** [iorate/ublacklist#430]
## Expected Behavior

For uBlackList to be always active. I cant even reach reddit as they have blacklisted so many vpn nodes so there is no reason to ever have reddit links populate in search results especially if I cant view their content. It really sucks because sometimes the first 5 results will be links to reddit.

## Actual Behavior

Currently I have to activate uBlackList for it to remove undesired sites from search results every time I open a browser. I have looked through the settings and cant find any way to add auto enable by default. Maybe I'm missing something? 

Maybe this is more of a feature request more than a bug report.

## Steps to Reproduce the Problem

1. Open Browser
1. use a search engine "DuckDuck" is what I generally use.
1. Then you have to select activate from the addons bar

## Specifications

- Browser: Firefox, Linux-Arch
- Other extensions/add-ons: Ublock, noScript, libRedirect and useragent switcher

- Search engine: DuckDuckGo
- Language: English
- Region: USA

**Comment by Freso** [comment]
Just to make sure: did you enable DuckDuckGo in the options page (as per [“Other search engines”](https://iorate.github.io/ublacklist/docs/advanced-features#other-search-engines))?

**Comment by tetoNidan** [comment]
Yes I do have DDG enabled in the 'other search engines'. I enabled all of the options for search engines even though I really dont use them.

**Comment by aug-dev** [comment]
I can attest to this issue.

I just tested and it seems to be mostly a problem with the Firefox version of the extension.

When Firefox is launched, the extension isn't active by default on any search engine besides Google:

![image](https://github.com/iorate/ublacklist/assets/109992671/bceebd60-56db-456b-b7b6-d02fc5d72584)

This happens even if you've already enabled the search engines through the options page.

Once you activate one them (e.g. for DuckDuckGo), all other secondary extensions seem to become active too.

I tested on Firefox 124.0.1 (flatpak version), and on Chromium.

I'll look further into this and see if can find a possible solution.

**Comment by NanoExplorer** [comment]
This is especially problematic on Firefox Mobile, since you don't always know when the OS has closed the browser in the background.

**Comment by NanoExplorer** [comment]
This comment would appear to be relevant, but there are no changes in it for some reason. 5b88625cb8afbfd2513bab33323a6c7e16ad70b8

**Comment by iorate** [comment]
I can no longer reproduce this. The content script registration method has changed since this was reported (`browser.scripting.registerContentScripts` instead of `browser.contentScripts.register`), which may have resolved the issue.

Closing for now. If you still experience this on the latest version, please open a new discussion.
